File: src/main/java/org/shark/example/service/mysql/TaskSchedule.java
package org.shark.example.service.mysql;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.shark.example.dao.repository.TaskHistoryRepository;
import org.shark.example.dao.repository.TaskRepository;
import org.shark.example.dao.repository.pojo.TaskDo;
import org.shark.example.dao.repository.pojo.TaskHistoryDo;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.UUID;

@Slf4j
@RequiredArgsConstructor
@Component
public class TaskSchedule {

    private final TaskRepository taskRepository;
    private final TaskHistoryRepository taskHistoryRepository;

    @Scheduled(initialDelay = 10000, fixedRate = 10000)
    public void doTask() {
        String workId = UUID.randomUUID().toString();
        taskRepository.updateWorkId(workId);
        TaskDo taskDo = taskRepository.findByWorkId(workId);
        if(taskDo == null) {
            return;
        }
        taskDo.setStatus(1);
        taskRepository.save(taskDo);
        TaskHistoryDo taskHistoryDo = new TaskHistoryDo();
        taskHistoryDo.setTaskId(taskDo.getId());
        taskHistoryDo.setWorkId(workId);
        taskHistoryDo.setStatus(1);
        taskHistoryRepository.save(taskHistoryDo);
    }
}

File: src/main/kotlin/org/shark/example/service/mysql/TaskSchedule.kt
package org.shark.example.service.mysql

import org.shark.example.dao.repository.TaskHistoryRepository
import org.shark.example.dao.repository.TaskRepository
import org.shark.example.dao.repository.pojo.TaskDo
import org.shark.example.dao.repository.pojo.TaskHistoryDo
import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class TaskSchedule(
    private val taskRepository: TaskRepository,
    private val taskHistoryRepository: TaskHistoryRepository
) {
    private val log = LoggerFactory.getLogger(TaskSchedule::class.java)

    @Scheduled(initialDelay = 10000, fixedRate = 10000)
    fun doTask() {
        val workId = UUID.randomUUID().toString()
        taskRepository.updateWorkId(workId)
        val taskDo: TaskDo? = taskRepository.findByWorkId(workId)
        if (taskDo == null) {
            return
        }
        taskDo.status = 1
        taskRepository.save(taskDo)
        val taskHistoryDo = TaskHistoryDo()
        taskHistoryDo.taskId = taskDo.id
        taskHistoryDo.workId = workId
        taskHistoryDo.status = 1
        taskHistoryRepository.save(taskHistoryDo)
    }
}
